lib/include/tree_sitter/api.h

```diff
@@ -79,6 +79,15 @@ typedef struct TSInput {
   TSInputEncoding encoding;
 } TSInput;
 
+typedef struct TSParseState {
+  void *payload;
+} TSParseState;
+
+typedef struct TSParseOptions {
+  void *payload;
+  bool (*interrupt_callback)(TSParseState *state);
```

Comment:
What do you think of calling this something like `progress_callback`, since it's called multiple times during normal parsing?

And actually, we could pass your current byte offset to let you track the progress. That's a good use case for `TSParserState` struct!

Reply:
That's really useful! Can we apply this to the `TSQueryCursorState` too with the current node's start byte?

Reply:
Yeah, that's a good idea. Might have to grab the tree-cursor's node or something.

Reply:
Updated :grin: